src/Razor/test/Microsoft.AspNetCore.Razor.LanguageServer.Test/Refactoring/CodeActionResolutionEndpointTest.cs

```diff
@@ -0,0 +1,109 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Razor.LanguageServer.Refactoring;
+using Microsoft.AspNetCore.Razor.Test.Common;
+using OmniSharp.Extensions.LanguageServer.Protocol.Models;
+using Xunit;
+using System.Threading;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.AspNetCore.Razor.LanguageServer.Test.Refactoring
+{
+    public class CodeActionResolutionEndpointTest : LanguageServerTestBase
+    {
+        [Fact]
+        public async Task Handle_Resolve()
+        {
+            // Arrange
+            var codeActionEndpoint = new CodeActionResolutionEndpoint(new RazorCodeActionResolver[] {
+                new MockCodeActionResolver("Test"),
+            }, LoggerFactory);
+            var request = new RazorCodeActionResolutionParams()
+            {
+                Action = "Test",
+                Data = null
+            };
+
+            // Act
+            var workspaceEdit = await codeActionEndpoint.Handle(request, default);
+
+            // Assert
+            Assert.NotNull(workspaceEdit);
+        }
+
+        [Fact]
+        public async Task Handle_ResolveMultipleA()
```

Comment:
A is a detail of the specific test. I'd change it to be something like 
```suggestion
        public async Task Handle_ResolveMultipleProviders_FirstMatches
```

Ditto for other tests here